src/main/java/me/moodcat/database/entities/Room.java

```diff
@@ -1,131 +1,109 @@
+
+import static javax.persistence.CascadeType.ALL;
+import static javax.persistence.FetchType.EAGER;
+import static javax.persistence.FetchType.LAZY;
 
 /**
  * A representation for a room, the room mainly supplies which song is currently listened by users
  * of the room and then position of the room.
+ *
+ * @author Jaap Heijligers
  */
 @Data
 @Entity
 @Table(name = "room")
 @ToString(of = {
-        "id", "song", "name"
+        "id",
 })
 @EqualsAndHashCode(of = "id")
 public class Room {
 
     /**
      * The unique identifier for the room.
-     *
-     * @param id
-     *            The id to set.
-     * @return The id of this room.
      */
     @Id
     @GeneratedValue(strategy = GenerationType.IDENTITY)
     @Column(name = "id", nullable = false)
     private Integer id;
 
+    @Embedded
+    @JsonIgnore
+    private VAVector vaVector;
+
     /**
      * The current song of the room.
-     *
-     * @param song
-     *            The song to set.
-     * @return The currently played song in this room.
      */
-    @ManyToOne(optional = false)
+    @ManyToOne
     @JoinColumn(name = "currentSong")
-    private Song song;
+    @JsonProperty("song")
+    private Song currentSong;
 
     /**
-     * The current position of the song in milliseconds.
-     *
-     * @param position
-     *            The position of this song.
-     * @return The position of the song in the queue.
+     * Songs to be played.
      */
-    private Integer position;
+    @ManyToMany(fetch = EAGER)
+    @JoinTable(name = "room_play_queue", joinColumns = {
+            @JoinColumn(name = "room_id", referencedColumnName = "id")
+    }, inverseJoinColumns = {
+            @JoinColumn(name = "song_id", referencedColumnName = "id")
+    })
+    private List<Song> playQueue;
 
     /**
      * The name of the room.
-     *
-     * @param name
-     *            The name to set.
-     * @return The name of this room.
      */
     @Column(name = "name")
     private String name;
 
     /**
-     * The time of the {@link #song} in order to 'jump' right into listening.
-     * This is not persisted in the database due to the high rate of updating
-     *
-     * @param time
-     *            The time the song is played on.
-     * @return The time that this song is played at.
+     * The songs recently played in the roomProvider<ChatDAO> chatDAOProvider.
      */
-    private int time;
+    @ManyToMany(fetch = LAZY)
+    @JsonIgnore
+    @JoinTable(name = "room_play_history", joinColumns = {
+            @JoinColumn(name = "room_id", referencedColumnName = "id")
+    }, inverseJoinColumns = {
+            @JoinColumn(name = "song_id", referencedColumnName = "id")
+    })
+    private List<Song> playHistory;
 
     /**
-     * The arousal value of this room.
-     *
-     * @param arousal
-     *            The arousal component of this room.
-     * @return The arousal of this room.
+     * Development flag to temporarily repeat the current song in a room.
```

Comment:
Could you explain this? Why would the arousal value be a "flag", and why would it  repeat the current song?